File: backend/app/schemas/keys.py
"""Pydantic schemas for API keys."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class KeyCreate(BaseModel):
    """Request body for POST /keys."""

    name: str


class KeyRead(BaseModel):
    """Response schema for listing keys (no secret)."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    key_prefix: str
    created_at: datetime
    last_used_at: datetime | None = None
    is_active: bool


class KeyCreated(BaseModel):
    """Response schema for newly created key (shows full key once)."""

    id: uuid.UUID
    name: str
    key: str
    key_prefix: str
    created_at: datetime

File: frontend/src/types/keys.ts
/** Shared TypeScript types for API keys. */

export interface APIKey {
    id: string;
    name: string;
    key_prefix: string;
    created_at: string;
    last_used_at: string | null;
    is_active: boolean;
}

export interface CreateKeyRequest {
    name: string;
}

export interface CreatedKey {
    id: string;
    name: string;
    key: string;
    key_prefix: string;
    created_at: string;
}
